x-pack/plugins/fleet/server/services/epm/packages/package_verification.ts

```diff
@@ -46,28 +57,61 @@ export async function _readGpgKey(): Promise<openpgp.Key | undefined> {
   return key;
 }
 
-export interface PackageVerificationResult {
-  isVerified: boolean;
-  keyId: string;
+export async function verifyPackageArchiveSignature({
+  pkgName,
+  pkgVersion,
+  pkgArchiveBuffer,
+  logger,
+}: {
+  pkgName: string;
+  pkgVersion: string;
+  pkgArchiveBuffer: Buffer;
+  logger: Logger;
+}): Promise<PackageVerificationResult> {
+  const verificationKey = await getGpgKeyOrUndefined();
+  const result: PackageVerificationResult = { verificationStatus: 'unknown' };
+  if (!verificationKey) {
+    logger.warn(`Not performing package verification as no local verification key found`);
+    return result;
+  }
+  const pkgArchiveSignature = await Registry.getPackageArchiveSignatureOrUndefined({
+    pkgName,
+    pkgVersion,
+    logger,
+  });
+
+  if (!pkgArchiveSignature) {
+    logger.warn(`Not performing package verification as package has no signature`);
```

Comment:
Let's reword this to `Package ${pkgName}-${pkgVersion} has no corresponding signature. Skipping verification.`